pulser-core/pulser/register/weight_maps.py

```diff
@@ -72,7 +74,7 @@ def sorted_weights(self) -> np.ndarray:
         return cast(np.ndarray, np.array(self.weights)[sorting])
 
     def get_qubit_weight_map(
-        self, qubits: Mapping[QubitId, np.ndarray]
+        self, qubits: Mapping[QubitId, pm.AbstractArray]
```

Comment:
From a user perspective, wouldn't `AbstractArrayLike` be better suited (the current implementation is introducing a breaking change, we can't provide np.ndarray anymore) ?

Reply:
From your comment above,  I guess you can even go for `ArrayLike` ?

Reply:
Yes, that's what I did :)